pkg/authorization/controller/defaultrolebindings/defaultrolebindings.go

```diff
@@ -0,0 +1,198 @@
+	roleBindingLister rbaclisters.RoleBindingLister
+	roleBindingSynced cache.InformerSynced
+	namespaceLister   corelisters.NamespaceLister
+	namespaceSynced   cache.InformerSynced
+
+	syncHandler func(namespace string) error
+	queue       workqueue.RateLimitingInterface
+}
+
+// NewDefaultRoleBinding creates a new controller
+func NewDefaultRoleBindingsController(roleBindingInformer rbacinformers.RoleBindingInformer, namespaceInformer coreinformers.NamespaceInformer, roleBindingClient rbacclient.RoleBindingsGetter) *DefaultRoleBindingController {
+	c := &DefaultRoleBindingController{
+		roleBindingClient: roleBindingClient,
+
+		roleBindingLister: roleBindingInformer.Lister(),
+		roleBindingSynced: roleBindingInformer.Informer().HasSynced,
+		namespaceLister:   namespaceInformer.Lister(),
+		namespaceSynced:   namespaceInformer.Informer().HasSynced,
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ClusterRoleAggregator"),
+	}
+	c.syncHandler = c.syncNamespace
+
+	roleBindingInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: func(obj interface{}) bool {
+			metadata, err := meta.Accessor(obj)
+			if err != nil {
+				return false
+			}
+			return defaultRoleBindingNames.Has(metadata.GetName())
+		},
+		Handler: cache.ResourceEventHandlerFuncs{
+			DeleteFunc: func(uncast interface{}) {
+				metadata, err := meta.Accessor(uncast)
+				if err == nil {
+					c.queue.Add(metadata.GetNamespace())
+					return
+				}
+
+				tombstone, ok := uncast.(cache.DeletedFinalStateUnknown)
+				if !ok {
+					utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %#v", uncast))
+					return
+				}
+				metadata, err = meta.Accessor(tombstone.Obj)
+				if err != nil {
+					utilruntime.HandleError(err)
+					return
+				}
+				c.queue.Add(metadata.GetNamespace())
+			},
+		},
+	})
+	namespaceInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			metadata, err := meta.Accessor(obj)
+			if err != nil {
+				utilruntime.HandleError(err)
+				return
+			}
+			c.queue.Add(metadata.GetName())
+		},
+	})
+	return c
+}
+
+func (c *DefaultRoleBindingController) syncNamespace(namespaceName string) error {
+	namespace, err := c.namespaceLister.Get(namespaceName)
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	if namespace.DeletionTimestamp != nil {
+		return nil
+	}
+
+	roleBindings, err := c.roleBindingLister.RoleBindings(namespaceName).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	errs := []error{}
+	desiredRoleBindings := bootstrappolicy.GetBootstrapServiceAccountProjectV1RoleBindings(namespaceName)
+	for i := range desiredRoleBindings {
+		desiredRoleBinding := desiredRoleBindings[i]
+		found := false
+		for _, existingRoleBinding := range roleBindings {
+			if existingRoleBinding.Name == desiredRoleBinding.Name {
+				found = true
+				break
+			}
+		}
+		if found {
```

Comment:
> It is still possible to opt out if you really want to break openshift features in your project by removing the subjects from the binding (I only match on name).

This is an interesting choice.  Why not the standard pattern of auto reconcile unless you have the protect annotation?

Reply:
> This is an interesting choice. Why not the standard pattern of auto reconcile unless you have the protect annotation?

It's just unnecessary. Presence or absence is enough.  Figure that you'd have to create it anyway to annotate it.